### ISSUE TITLE
Basic test using Firefox on Window occasionally fails (selenium-java 4.0.0-rc-1)

### ISSUE THREAD

**Issue by bonigarcia** [SeleniumHQ/selenium#9842]
## 🐛 Bug Report

A very basic [test](https://github.com/bonigarcia/selenium-webdriver-java/blob/master/selenium-webdriver-junit4/src/test/java/io/github/bonigarcia/webdriver/junit4/ch2/helloworld/HelloWorldFirefoxJUnit4Test.java) using Firefox and executed on Windows [fails](https://github.com/bonigarcia/selenium-webdriver-java/runs/3637003836?check_suite_focus=true) from time to time.

## To Reproduce

I run this [test](https://github.com/bonigarcia/selenium-webdriver-java/blob/master/selenium-webdriver-junit4/src/test/java/io/github/bonigarcia/webdriver/junit4/ch2/helloworld/HelloWorldFirefoxJUnit4Test.java) in `windows-latest` (Windows Server 2019) on GitHub Actions. Randomly, it fails showing the following error traces (see the complete [execution logs](https://github.com/bonigarcia/selenium-webdriver-java/runs/3637003836?check_suite_focus=true)):

```
Sep 17, 2021 10:05:34 PM org.openqa.selenium.os.OsProcess destroy
INFO: Unable to drain process streams. Ignoring but the exception being swallowed follows.
org.apache.commons.exec.ExecuteException: The stop timeout of 2000 ms was exceeded (Exit value: -559038737)
	at org.apache.commons.exec.PumpStreamHandler.stopThread(PumpStreamHandler.java:295)
	at org.apache.commons.exec.PumpStreamHandler.stop(PumpStreamHandler.java:181)
	at org.openqa.selenium.os.OsProcess.destroy(OsProcess.java:135)
	at org.openqa.selenium.os.CommandLine.destroy(CommandLine.java:152)
	at org.openqa.selenium.remote.service.DriverService.stop(DriverService.java:278)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:151)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:547)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:244)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:166)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:205)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:193)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:171)
	at io.github.bonigarcia.webdriver.junit4.ch2.helloworld.HelloWorldFirefoxJUnit4Test.setup(HelloWorldFirefoxJUnit4Test.java:46)
	...

Sep 17, 2021 10:05:34 PM org.openqa.selenium.os.OsProcess destroy
SEVERE: Unable to kill process java.lang.ProcessImpl@48d7ad8b
Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 141.686 sec <<< FAILURE!
test(io.github.bonigarcia.webdriver.junit4.ch2.helloworld.HelloWorldFirefoxJUnit4Test)  Time elapsed: 140.584 sec  <<< ERROR!
org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: No connection could be made because the target machine actively refused it. (os error 10061)
Build info: version: '4.0.0-rc-1', revision: 'bc5511cbda'
System info: host: 'fv-az159-878', ip: '10.1.0.165', os.name: 'Windows Server 2019', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_302'
Driver info: org.openqa.selenium.firefox.FirefoxDriver
Command: [null, newSession {capabilities=[Capabilities {acceptInsecureCerts: true, browserName: firefox, moz:debuggerAddress: true, moz:firefoxOptions: {}}], desiredCapabilities=Capabilities {acceptInsecureCerts: true, browserName: firefox, moz:debuggerAddress: true, moz:firefoxOptions: {}}}]
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:126)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:84)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:62)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:156)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:164)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:139)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:547)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:244)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:166)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:205)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:193)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:171)
	at io.github.bonigarcia.webdriver.junit4.ch2.helloworld.HelloWorldFirefoxJUnit4Test.setup(HelloWorldFirefoxJUnit4Test.java:46)
	...
```

## Expected behavior

This test should pass. The same test is being executed also in `ubuntu-latest` (Ubuntu 20.04) and `macos-latest` (macOS 10.15) in GitHub Actions without any problem.

## Test script or set of commands reproducing this issue

1. Run this [test](https://github.com/bonigarcia/selenium-webdriver-java/blob/master/selenium-webdriver-junit4/src/test/java/io/github/bonigarcia/webdriver/junit4/ch2/helloworld/HelloWorldFirefoxJUnit4Test.java) on Windows virtual environment in GitHub Actions.
2. Repeat until you get the error.

I tried to find a solution for the problem without luck. I found that the underlying cause, which seems to `No connection could be made because the target machine actively refused it. (os error 10061)`, is related to a socket connection problem with winsock. In similar cases, the recommended solution is to retry the connection (see [example](https://stackoverflow.com/questions/2972600/no-connection-could-be-made-because-the-target-machine-actively-refused-it)). Is this a possible solution for this issue?

## Environment

OS: Windows Server 2019
Browser: Firefox
Browser version: 92.0
Browser Driver version: geckodriver 0.30.0
Language Bindings version: selenium-java 4.0.0-rc-1
Selenium Grid version (if applicable): -

**Comment by diemol** [comment]
This seems like Firefox having issues starting up and then what the binding tries to do is to stop the browser driver to leave a clean environment. 

Even if we find a way to retry the connection, the browser or the browser driver has already crashed, which then could lead to a scenario where resources are wasted. 

If this is only reproducible after several repetitions only in GH actions, it indicates an issue with that environment. Is it possible to reprouce it in a different Windows environment? Maybe it could be worth to report this to [Mozilla](https://bugzilla.mozilla.org/home)?

**Comment by bonigarcia** [comment]
I only managed to reproduce in GitHub Actions. But I suppose this problem might occur in different environment. I said that because it seems it is a recurring problem, at least there are quite a lot of related questions in [StackOverflow](https://stackoverflow.com/search?q=10061%20selenium).

In any case, you might be right, and it seems it is not a Selenium issue, but geckodriver. I just found an open issue about it in [Bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1630132). Thanks for your support @diemol.
